cpg-core/src/main/java/de/fraunhofer/aisec/cpg/graph/statements/expressions/BinaryOperator.java

```diff
@@ -245,4 +242,27 @@ public boolean equals(Object o) {
   public int hashCode() {
     return super.hashCode();
   }
+
+  @Nullable
+  @Override
+  public AssignmentTarget getTarget() {
+    // We only want to supply a target if this is an assignment
+    return isAssignment()
+        ? (lhs instanceof AssignmentTarget ? (AssignmentTarget) lhs : null)
+        : null;
+  }
+
+  @Nullable
+  @Override
+  public Expression getValue() {
+    return isAssignment() ? rhs : null;
+  }
+
+  public boolean isAssignment() {
+    // TODO(oxisto): We need to discuss, if the other operators are also assignments and if we
+    // really want them
+    return this.operatorCode.equals(
```

Comment:
Why is `==` an assignment? Isn't that pretty much the only operator which is not an assignment? I guess you meant `=` or maybe wanted to negate the condition?

Reply:
Typo, should be `= `